### PR TITLE
Nancy Dependency Scan in CI Pipeline

### PR DESCRIPTION
**Pull request by EasterTheBunny**
The change here adds Nancy scans to the CI pipeline just after unit tests to ensure dependencies are scanned for vulnerabilities on a regular basis.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -2,43 +2,48 @@ module github.com/smartcontractkit/ocr2keepers
...
+	github.com/go-echarts/go-echarts/v2 v2.2.5
+	github.com/google/uuid v1.3.0
+	github.com/pkg/errors v0.9.1
 	github.com/shopspring/decimal v1.3.1
-	github.com/smartcontractkit/libocr v0.0.0-20221209172631-568a30f68407
-	github.com/spf13/pflag v1.0.3
-	github.com/stretchr/testify v1.7.2
-	go.uber.org/multierr v1.6.0
-	golang.org/x/crypto v0.0.0-20221012134737-56aed061732a
+	github.com/smartcontractkit/libocr v0.0.0-20230112103111-f100435787c8
```
Comment (infiloop2) [inline go.mod:17]: what is the version used in CL repo?

File: .nancy-ignore [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# The go-libp2p dependency is not used directly in this plugin so the following
```
Comment (infiloop2) [inline .nancy-ignore:1]: i'm not sure if we're even using libp2p for ocr anymore. is this copied from CL repo?

File: go.mod (hunk 1 above)
Comment (EasterTheBunny) [reply go.mod:17]: The chainlink node uses commit 568a30f but this repo uses the latest commit of f100435.

File: .nancy-ignore (hunk 2 above)
Comment (EasterTheBunny) [reply .nancy-ignore:1]: No. I made this comment to indicate that this repo only really uses libocr for types and interfaces and doesn't use libp2p directly. The libocr repo may still have this as a dependency even if the chainlink repo uses a different implementation. OCR2 allows for an arbitrary networking layer as it defines an expected interface, not a distinct type.

File: go.mod (hunk 1 above)
Comment (EasterTheBunny) [reply go.mod:17]: So not use the latest?

Comment (infiloop2) [reply go.mod:17]: yep

Comment (EasterTheBunny) [reply go.mod]: done

**Review by infiloop2 — APPROVED** [review]

**Pull request merged**